src/sagas/reown.js

```diff
@@ -0,0 +1,861 @@
+          data,
+        });
+      };
+
+      walletKit.on(eventName, listener);
+      listenerMap.set(eventName, listener);
+    };
+
+    addListener('session_request');
+    addListener('session_proposal');
+    addListener('session_delete');
+    addListener('disconnect');
+
+    return () => listenerMap.forEach((
+      listener,
+      eventName,
+    ) => walletKit.removeListener(eventName, listener));
+  });
+
+  try {
+    while (true) {
+      const message = yield take(channel);
+
+      yield put({
+        type: message.type,
+        payload: message.data,
+      });
+    }
+  } catch (e) {
+    log.error(e);
+  } finally {
+    if (yield cancelled()) {
+      // When we close the channel, it will remove the event listener
+      channel.close();
+    }
+  }
+}
+
+export function* clearSessions() {
+  const { walletKit } = getGlobalReown();
+  if (!walletKit) {
+    log.debug('Tried to get reown client in clearSessions but walletKit is undefined.');
+    return;
+  }
+
+  const activeSessions = yield call(() => walletKit.getActiveSessions());
+
+  for (const key of Object.keys(activeSessions)) {
+    yield call(() => walletKit.disconnectSession({
+      topic: activeSessions[key].topic,
+      reason: {
+        code: ERROR_CODES.USER_DISCONNECTED,
+        message: '',
+      },
+    }));
+  }
+
+  yield call(refreshActiveSessions);
+}
+
+function* requestsListener() {
+  const requestsChannel = yield actionChannel('REOWN_SESSION_REQUEST');
+
+  let action;
+  while (true) {
+    try {
+      action = yield take(requestsChannel);
+      yield call(processRequest, action);
+    } catch (error) {
+      log.error('Error processing request.', error);
+      yield put(onExceptionCaptured(error));
+    }
+  }
+}
+
+export function* processRequest(action) {
+  const { payload } = action;
+  const { params } = payload;
+
+  const { walletKit } = getGlobalReown();
+  if (!walletKit) {
+    log.debug('Tried to get reown client in processRequest but walletKit is undefined.');
+    return;
+  }
+
+  const activeSessions = yield call(() => walletKit.getActiveSessions());
+  const requestSession = activeSessions[payload.topic];
+
+  if (!requestSession) {
+    log.error('Could not identify the request session, ignoring request.');
+    return;
+  }
+
+  const data = {
+    icon: get(requestSession.peer, 'metadata.icons[0]', null),
+    proposer: get(requestSession.peer, 'metadata.name', ''),
+    url: get(requestSession.peer, 'metadata.url', ''),
+    description: get(requestSession.peer, 'metadata.description', ''),
+    chain: get(requestSession.namespaces, 'hathor.chains[0]', ''),
+  };
+
+  try {
+    const wallet = getGlobalWallet();
+    let dispatch;
+    yield put((_dispatch) => {
+      dispatch = _dispatch;
+    });
+
+    const response = yield call(
+      handleRpcRequest,
+      params.request,
+      wallet,
+      data,
+      promptHandler(dispatch),
+    );
+
+    switch (response.type) {
```

Comment:
Shouldn't you add the sign message with address here as well?

Reply:
Yes!

Added the `MODAL_TYPES.MESSAGE_SIGNING_FEEDBACK` here as well.